crates/nu-command/src/system/run_external.rs

```diff
@@ -41,8 +44,128 @@ impl Command for External {
+        if is_cmd_internal_command(&name.item) {
+            use std::os::windows::process::CommandExt;
+
+            // The /D flag disables execution of AutoRun commands from registry.
+            // The /C flag followed by a command name instructs CMD to execute
+            // that command and quit.
+            command.args(["/D", "/C", &name.item]);
+            for arg in &args {
+                command.raw_arg(escape_cmd_argument(arg)?.as_ref());
+            }
+        } else {
+            command.args(args.into_iter().map(|s| s.item));
+        }
+        #[cfg(not(windows))]
+        command.args(args.into_iter().map(|s| s.item));
+
+        // Configure stdout and stderr. If both are set to `OutDest::Pipe`,
+        // we'll setup a pipe that merge two streams into one.
+        let stdout = stack.stdout();
+        let stderr = stack.stderr();
+        let merged_stream = if matches!(stdout, OutDest::Pipe) && matches!(stderr, OutDest::Pipe) {
+            let (reader, writer) = os_pipe::pipe()?;
+            command.stdout(writer.try_clone()?);
+            command.stderr(writer);
+            Some(reader)
+        } else {
+            command.stdout(Stdio::try_from(stdout)?);
+            command.stderr(Stdio::try_from(stderr)?);
+            None
+        };
+
+        // Configure stdin. We'll try connecting input to the child process
+        // directly. If that's not possible, we'll setup a pipe and spawn a
+        // thread to copy data into the child process.
+        let data_to_copy_into_stdin = match input {
+            PipelineData::ByteStream(stream, metadata) => match stream.into_stdio() {
+                Ok(stdin) => {
+                    command.stdin(stdin);
+                    None
+                }
+                Err(stream) => {
+                    command.stdin(Stdio::piped());
+                    Some(PipelineData::ByteStream(stream, metadata))
+                }
+            },
+            PipelineData::Empty => {
+                command.stdin(Stdio::inherit());
+                None
+            }
+            value => {
+                command.stdin(Stdio::piped());
+                Some(value)
+            }
+        };
+
+        // Spawn the child process. On Unix, also put the child process to
+        // foreground if we're in an interactive session.
+        #[cfg(windows)]
+        let mut child = ForegroundChild::spawn(command)?;
+        #[cfg(unix)]
+        let mut child = ForegroundChild::spawn(
+            command,
+            engine_state.is_interactive,
+            &engine_state.pipeline_externals_state,
+        )?;
+
+        // If we need to copy data into the child process, do it now.
+        if let Some(data) = data_to_copy_into_stdin {
+            let stdin = child.as_mut().stdin.take().expect("stdin is piped");
+            let engine_state = engine_state.clone();
+            let stack = stack.clone();
+            thread::spawn(move || {
```

Comment:
I think it'd be good to attach a name to the thread, like how it does originally:
```rust
thread::Builder::new()
       .name("external stdin worker".to_string())
       .spawn(..)
```

Reply:
As mentioned in the [documentation](https://doc.rust-lang.org/std/thread/struct.Builder.html#method.name), the name is only used for panic messages. But our thread never panics, so attaching a name doesn't do anything.

Reply:
Why would the thread never panic? I think it's possible for the inner `dyn Read` in a `ByteStream`,  for the `ListStream` passed to the `table` command, and for the `table` command itself to panic.

Reply:
I meant that we're not relying on the thread panicking as part of our error handling strategy. Of course, the thread may panic if there's a bug somewhere, so attaching a name helps with debugging. I'll do it then.